### ISSUE TITLE
[bug]: some tracks not refreshed in api.reload()

### ISSUE THREAD

**Issue by manzt** [higlass/higlass#1084]
_Commit_: #1062 

It seems the implementation in #1062 is lacking, and I've come across some strange behavior I'm not able to sort out. context: I am trying to use this feature for dynamic/filtered tilesets in higlass-python/hg.

#### Steps to reproduce
1. checkout https://github.com/manzt/higlass/tree/reload-bug, and `npm start`.

#### Observed behavior

```javascript
setInterval(() => 
  hgApi.reload([
    { trackId: 'line1', viewId: 'aa' }, // not refreshed
    { trackId: 'line2', viewId: 'aa' },
    { trackId: 'line4', viewId: 'aa' },
  ]),
  3000,
);
```

https://user-images.githubusercontent.com/24403730/153774015-e1977ede-d7d6-4584-8a2a-04fa2ba312d9.mov

refreshes track data for `line2`, `line4` but not `line1`. 

**NOTE** if I reverse the order of identifiers (`line4`, `line2`, `line1`), the _first_ track is skipped again and now only `line2` and `line1` are rendered.

```javascript
setInterval(() => 
  hgApi.reload([
    { trackId: 'line4', viewId: 'aa' }, // not refreshed
    { trackId: 'line2', viewId: 'aa' }, 
    { trackId: 'line1', viewId: 'aa' },
  ]),
  3000,
);
```

https://user-images.githubusercontent.com/24403730/153774173-ab2e7479-6f3f-4bdc-a6f2-98179170d511.mov

I have tried duplicating the first entry, in hopes of a work around but that also doesn't seem to work...

```javascript
setInterval(() => 
  hgApi.reload([
    { trackId: 'line4', viewId: 'aa' }, // not refreshed
    { trackId: 'line4', viewId: 'aa' }, // not refreshed
    { trackId: 'line2', viewId: 'aa' }, 
    { trackId: 'line1', viewId: 'aa' },
  ]),
  3000,
);
```
 
#### Expected behavior
-  All track targets should reload as specified.

I'm really at a loss for what is going on here, so hopefully someone might have some insight.

**Comment by pkerpedjiev** [comment]
Dang, you're right. It's super puzzling. What's even stranger is that the tiles for line1 and line2 are requested but not for line4 (see the network panel below).

![image](https://user-images.githubusercontent.com/2143629/154207832-48aa5088-767c-4a55-b858-ae12e1109361.png)

![image](https://user-images.githubusercontent.com/2143629/154207867-f331677e-e66f-4008-b919-ab9791a74a81.png)

**Comment by pkerpedjiev** [comment]
Just to confirm, when you see this issue the criteria for detecting refreshing is not just seeing a flicker on the screen, is it?

**Comment by pkerpedjiev** [comment]
It does look like the `drawTile` function gets called for each line track...

**Comment by pkerpedjiev** [comment]
When the mouse moves the flickering appears on the first track.

https://user-images.githubusercontent.com/2143629/154209117-a86bdc93-092c-4f66-bb2d-56dc2cca511f.mp4

**Comment by manzt** [comment]
> Just to confirm, when you see this issue the criteria for detecting refreshing is not just seeing a flicker on the screen, is it?

Yes, I believe I saw something similar in the network tab (plus the flickering). 

I'll be able to take a closer look tomorrow. I can probably set something up where I know the server data is changing so we should see visual changes for every track on the client.
